jdaviz/configs/imviz/plugins/coords_info/coords_info.py

```diff
@@ -493,9 +487,18 @@ def _image_viewer_update(self, viewer, x, y):
                     image, arr, x, y, viewer
                 )
 
-                # We don't want to convert for things like moment maps
-                if str(u.Unit(unit).physical_type) not in ("spectral flux density",
-                                                           "surface brightness"):
+                # We don't want to convert for things like moment maps, so check physical type
+                # If unit is flux per pix2, the type will be 'unknown' rather
+                # than surface brightness, so have to multiply the pix2 part out
+                # and check if the numerator is a spectral flux density
+                if check_if_unit_is_per_solid_angle(unit, return_unit=True) == u.pix*u.pix:
+                    un = u.Unit(unit) * u.pix*u.pix
+                    physical_type = un.physical_type
+                else:
+                    physical_type = u.Unit(unit).physical_type
+
+                if str(physical_type) not in ("spectral flux density",
+                                              "surface brightness"):
                     skip_spectral_density_eqv = True
```

Comment:
is this all just to skip the spectral density equivalency when the image units are per pix**2?  Do they hurt to just pass anyways?

Reply:
It's not to skip passing it, its that it just won't work. You cant compare Jy/pix2 to erg/cm/s/pix2 using the spectral density equivalency, for some reason. You can compare Jy and erg/cm/s using that equiv, but not with pix2 in the denominators. This is the issue with some other conversions as well, which require a workaround outside of passing in equivalencies

Reply:
ok, we can add this to the "generalize equivalency handling" follow-up then?